includes/admin/class-wc-admin-profile.php

```diff
@@ -202,6 +202,10 @@ public function add_customer_meta_fields( $user ) {
 		 * @param int $user_id User ID of the user being saved
 		 */
 		public function save_customer_meta_fields( $user_id ) {
+			if ( ! apply_filters( 'woocommerce_current_user_can_edit_customer_meta_fields', current_user_can( 'manage_woocommerce' ), get_userdata( $user ) ) ) {
```

Comment:
This should be using `$user_id`. Maybe we should just pass ID to both filters for ease and to avoid needing to load userdata here?